src/Database/Expression/TupleComparison.php

```diff
@@ -96,7 +96,7 @@ protected function _stringifyValues($generator)
             if ($isMulti) {
                 $bound = [];
                 foreach ($value as $k => $val) {
-                    $valType = $multiType ? $type[$k] : $type;
+                    $valType = $multiType && isset($type[$i]) ? $type[$k] : $type;
```

Comment:
Why do you check for `$i` when the read is using `$k`?

Reply:
Just stupid copy/paste...